### PR TITLE
Relax strictness of Quoting Strings with Single Quotes

### LINKED ISSUE #13531: Escape quotes not working
Challenge [Quoting Strings with Single Quotes](https://www.freecodecamp.com/challenges/quoting-strings-with-single-quotes#?solution=%0Avar%20myStr%20%3D'%3Ca%20href%3D%22http%3A%2F%2Fwww.example.com%22%20%20target%3D%20%22_blank%22%3ELink%3C%2Fa%3E'%3B%0A%0A%0A) has an issue.
User Agent is: <code>Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/56.0.2924.87 Safari/537.36</code>.
Please describe how to reproduce this issue, and include links to screenshots if possible.

My code:
```javascript

var myStr ='<a href="http://www.example.com"  target= "_blank">Link</a>';



```

### PR DESCRIPTION
**Pull request by IanTeo**
#### Pre-Submission Checklist

- [x] Your pull request targets the `staging` branch of freeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.com/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change

- [x] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [x] Tested changes locally.
- [x] Closes currently open issue (replace XXXX with an issue no): Closes #13531

#### Description
* Added "\s*" to multiple areas in the test to reduce the strictness of the [Quoting Strings with Single Quotes](https://www.freecodecamp.com/challenges/quoting-strings-with-single-quotes) test

### CODE REVIEW

File: seed/challenges/02-javascript-algorithms-and-data-structures/basic-javascript.json [hunk 1]
```diff
@@ -1084,7 +1084,7 @@
         "/* head */ 'use strict'; /* solution */ var myStr = '<a href=\"http://www.example.com\" target=\"_blank\">Link</a>'; /* tail */ (function() { return \"myStr = \" + myStr; })();"
       ],
       "tests": [
-        "assert(!/\\\\/g.test(code) && myStr === '<a href=\"http://www.example.com\" target=\"_blank\">Link</a>', 'message: Remove all the <code>backslashes</code> (<code>\\</code>)');",
+        "assert(!/\\\\/g.test(code) && myStr.match('\\\\s*<a href=\"http://www.example.com\"\\\\s*target=\"_blank\">\\\\s*Link\\\\s*</a>\\\\s*'), 'message: Remove all the <code>backslashes</code> (<code>\\</code>)');",
```
Comment (Greenheart) [inline seed/challenges/02-javascript-algorithms-and-data-structures/basic-javascript.json]: Do you think we should be precautions and allow spaces around for example the `=` characters as well?

The following string will not pass the tests at this time. Note the space after `target=`
```js
var myStr ='<a href="http://www.example.com" target= "_blank">Link</a>';
```

**Review by Greenheart — COMMENTED** [review]
@IanTeo Thanks for working on this! :smile:

File: seed/challenges/02-javascript-algorithms-and-data-structures/basic-javascript.json (hunk 1 above)
Comment (jdeprey) [inline seed/challenges/02-javascript-algorithms-and-data-structures/basic-javascript.json]: I agree that spaces should be allowed around the `=`.  The original code posted for this issue 
`var myStr ='<a href="http://www.example.com"  target= "_blank">Link</a>';` will still fail the test due to that extra whitespace after `target=`.

**Comment by IanTeo** [comment]
I have added the spaces around the equals signs too

**Review by Greenheart — APPROVED** [review]
@IanTeo This works great! Thank you! :+1:

**Pull request merged**